flamingpy/simulations.py

```diff
@@ -286,25 +242,16 @@ def run_ec_simulation(
     else:
         raise ValueError(f"Decoder {params['decoder']} is either invalid or not yet implemented.")
 
-    # The Monte Carlo simulations
-    code = SurfaceCode
-    code_args = {key: params[key] for key in ["distance", "ec", "boundaries"]}
-
-    noise = noise_dict[params["noise"]]
-    if params.get("noise") == "iid":
-        if params.get("error_probability") is None:
-            raise ValueError("No argument `err_prob` found for `iid` noise.")
-        noise_args = {"error_probability": params.get("error_probability")}
-    else:
-        noise_args = {key: params[key] for key in ["delta", "p_swap"]}
+    noise = str_dict[params["noise"]]
+    code = str_dict[params["code"]]
 
-    decoder = params["decoder"]
     args = {
         "trials": params["trials"],
         "code": code,
-        "code_args": code_args,
+        "code_args": l_eval(params["code_args"]),
         "noise": noise,
```

Comment:
```suggestion
        "noise": eval[params["noise"]),
```

Reply:
This is not needed, for consistency we have a `str_dict`, which translates all string terms including `noise` from the string form to an object as needed. See L248.